File: sumRange.js
/**
 * @constructor
 * @param {number[]} nums
 */
var NumArray = function(nums) {
  this.nums = nums;
};

/**
 * @param {number} i
 * @param {number} j
 * @return {number}
 */
NumArray.prototype.sumRange = function(i, j) {
  var sum = 0;
  for (var index = i; index <= j; index++) {
    sum += this.nums[index];
  }
  return sum;
};


/**
 * Your NumArray object will be instantiated and called as such:
 * var numArray = new NumArray(nums);
 * numArray.sumRange(0, 1);
 * numArray.sumRange(0, 2);
 */

 // var numArray = new NumArray([1, 3, 5, 8, 9]);
 // console.log(numArray.sumRange(0, 1));

File: sumRange.py
class NumArray(object):
    def __init__(self, nums):
        """
        initialize your data structure here.
        :type nums: List[int]
        """
        self.nums = nums

    def sumRange(self, i, j):
        """
        sum of elements nums[i..j], inclusive.
        :type i: int
        :type j: int
        :rtype: int
        """
        if i > j:
            return 0
        else:
            return self.nums[i] + self.sumRange(i + 1, j)
        


# Your NumArray object will be instantiated and called as such:
# numArray = NumArray(nums)
# numArray.sumRange(0, 1)
# numArray.sumRange(1, 2)

# numArray = NumArray([1, 3, 5, 6, 9])
# print(numArray.sumRange(0, 4))
